apis/field_error.go

```diff
@@ -73,10 +73,84 @@ func (fe *FieldError) ViaField(prefix ...string) *FieldError {
 				strings.Join(append(prefix, oldPath), "."))
 		}
 	}
+
+	newPaths = flattenIndexes(newPaths)
+
 	fe.Paths = newPaths
 	return fe
 }
 
+// ViaIndex is used to attach an index to the next via field provided.
+// For example, if a type recursively validates a parameter that has a collection:
+//  for i, c := range spec.Collection {
+//    if err := doValidation(c); err != nil {
+//      return err.ViaIndex(i).ViaField("collection")
+//    }
+//  }
+func (fe *FieldError) ViaIndex(index ...int) *FieldError {
```

Comment:
I feel like this would be more natural if combined given the inverted ordering:
```go
  return err.ViaIndex("collection", i)
```

Is Index the most idiomatic term?  Subscript?

Reply:
Index is the path into an array https://en.wikipedia.org/wiki/Index_notation

Subscript seems to imply the full path in the nth-dimensional array: http://mathworld.wolfram.com/Subscript.html

I think most people would understand index, and not subscript